Datacass 405 added additional unit tests

TheRealHaui opened the pull request:
<!--

Thank you for proposing a pull request. This template will guide you through the essential steps necessary for a pull request.
Make sure that:

-->

- [x ] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [ x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATACASS).
- [ x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [ x] You submit test cases (unit or integration tests) that back your changes.
- [x ] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java:
```diff
@@ -16,22 +16,36 @@
 package org.springframework.data.cassandra.repository.support;
 
-import static org.assertj.core.api.Assertions.*;
+import static junit.framework.TestCase.assertNotNull;
```
Please stick to using AssertJ.

TheRealHaui replied:
Done.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java:
```diff
@@ -16,22 +16,36 @@
+ * @author Michael Hausegger
  */
 public class BasicMapIdUnitTests {
+
```
Please use the [Spring Data formatting settings (imports and code style)](https://github.com/spring-projects/spring-data-cassandra/tree/master/etc).

TheRealHaui replied:
Rechecked desired code formatting settings in the used IDE are set.
Reformatted created and/or changed code using the desired code formatting settings.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java:
```diff
@@ -41,4 +55,471 @@ public void testMapConstructor() {
+		BasicMapId basicMapId = new BasicMapId();
+		basicMapId.putAll(basicMapId);
+
+		assertEquals(0, basicMapId.size());
```
Please use AssertJ (see above).

TheRealHaui replied:
Done.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java:
```diff
@@ -41,4 +55,471 @@ public void testMapConstructor() {
+
+
+	@Test
+	public void testOne() throws Exception {
```
`testOne` doesn't express the test subject. Use more expressive test names.

TheRealHaui replied:
Done.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java:
```diff
@@ -41,4 +55,471 @@ public void testMapConstructor() {
+	public void testOne() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		basicMapId.putAll(basicMapId);
```
What's the subject of this test?

TheRealHaui replied:
Verifies if the putAll method accepts itself as input parameter.
And furthermore verifies if the input parameter is allowed to be empty internally or not.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java:
```diff
@@ -41,4 +55,471 @@ public void testMapConstructor() {
+	public void testFour() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		HashMap<String, String> hashMap = new HashMap<String, String>();
```
Using `HashMap` variables seems odd. Why don't you stick with `Map`?

TheRealHaui replied:
As Unit Test methods in my opinion are not meant to get reused or inherited from on a regular basis I didn't suppose this to be a problem or odd.
However, I've changed the line.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java:
```diff
@@ -41,4 +55,471 @@ public void testMapConstructor() {
+	@Test
+	public void testThree() throws Exception {
+
+		BasicMapId basicMapId = (BasicMapId) BasicMapId.id("#xAD]-}", (Serializable) "#xAD]-}");
```
Why `#xAD]-}` instead of something more common, like `foo`, `bar`? Hard-to-read strings increase cognitive load.

TheRealHaui replied:
I have made very, very good experiences providing not awaited, uncommon strings to systems under test! Especially those with special meaning! For example if you think of security breaches and the long time in the future they are supposed to run.
However, I've replaced it with "a".

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/util/SpelUtilsTest.java:
```diff
@@ -0,0 +1,117 @@
+
+package org.springframework.data.cassandra.util;
+
+import static org.junit.Assert.*;
```
Please stick to using AssertJ.

TheRealHaui replied:
Done.

TheRealHaui replied:
Done

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/util/SpelUtilsTest.java:
```diff
@@ -0,0 +1,117 @@
+ * @see SpelUtils
+ **/
+public class SpelUtilsTest {
+
```
Please use the [Spring Data formatting settings (imports and code style)](https://github.com/spring-projects/spring-data-cassandra/tree/master/etc).

TheRealHaui replied:
Done.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/util/SpelUtilsTest.java:
```diff
@@ -0,0 +1,117 @@
+ * Unit tests for class {@link SpelUtils org.springframework.data.cassandra.util.SpelUtils}.
+ *
+ * @author Michael Hausegger
+ * @date 20.02.2017
```
We don't do `@see` and `@date` tags in our tests.

TheRealHaui replied:
Removed.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/util/SpelUtilsTest.java:
```diff
@@ -0,0 +1,117 @@
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+
+/**
+ * Unit tests for class {@link SpelUtils org.springframework.data.cassandra.util.SpelUtils}.
```
👍

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/util/SpelUtilsTest.java:
```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2016-2017 the original author or authors.
```
in this case, it's just `2017` (see the [Contribution guide](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc) for more details).

TheRealHaui replied:
Done.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/util/SpelUtilsTest.java:
```diff
@@ -0,0 +1,117 @@
+ * @author Michael Hausegger
+ * @date 20.02.2017
+ * @see SpelUtils
+ **/
```
`**/`? Why not just `*/`?

TheRealHaui replied:
I can't see the difference in this case.
Removed one  asterisk at the end of the comment.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/util/SpelUtilsTest.java:
```diff
@@ -0,0 +1,117 @@
+	@Test
+	public void testThree() throws Exception {
+
+		Class<String> clasz = String.class;
```
Any particular reason for the variable instead of inlining it?

TheRealHaui replied:
Convenience.
Variable was removed.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/util/SpelUtilsTest.java:
```diff
@@ -0,0 +1,117 @@
+	public void testOne() throws Exception {
+
+		SpelExpressionParser spelExpressionParser = SpelUtils.DEFAULT_PARSER;
+		String string = SpelUtils.evaluate((CharSequence) "cellEditor", (EvaluationContext) null, (ExpressionParser) spelExpressionParser);
```
Please use more expressive variable names. `result` might be a good fit.

mp911de replied:
Any reason why parameter casting is required?

TheRealHaui replied:
Done.

TheRealHaui replied:
I've incorporated the casting to ensure the test is working out of the box even if in the class under test further overloadings of the evaluate method get build in.
However I've removed the parameter castings in the tests.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/util/SpelUtilsTest.java:
```diff
@@ -0,0 +1,117 @@
+		assertFalse(spelParserConfiguration.isAutoGrowCollections());
+
+		assertEquals(SpelCompilerMode.OFF, spelParserConfiguration.getCompilerMode());
+		assertEquals(2147483647, spelParserConfiguration.getMaximumAutoGrowSize());
```
Use rather `Integer.MAX_VALUE` instead of a numeric literal. `MAX_VALUE` transports a different message than `2147483647` although both yield to the same integer value.

TheRealHaui replied:
Corrected.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/util/SpelUtilsTest.java:
```diff
@@ -0,0 +1,117 @@
+	@Test
+	public void testFour() throws Exception {
+
+		Class<Object> clasz = Object.class;
```
Any particular reason for the variable instead of inlining it?

TheRealHaui replied:
Convenience.
Corrected.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/util/SpelUtilsTest.java:
```diff
@@ -0,0 +1,117 @@
+	public void testSevenRaisesStringIndexOutOfBoundsException() throws Exception {
+
+		char[] charArray = new char[2];
+		Segment segment = new Segment(charArray, (-2826), 0);
```
Please don't use Swing types in here. `StringBuilder` might be the better choice.

TheRealHaui replied:
Removed the entire test.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java:
```diff
@@ -41,4 +55,471 @@ public void testMapConstructor() {
+
+
+	@Test
+	public void testTwentyNine() throws Exception {
```
Test methods should have a tight focus regarding their scope.

TheRealHaui replied:
I've split the test in two separate ones and have given both a meaningful name.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java:
```diff
@@ -41,4 +55,471 @@ public void testMapConstructor() {
+
+
+	@Test
+	public void testNine() throws Exception {
```
What does this method test?

TheRealHaui replied:
Method was removed.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java:
```diff
@@ -41,4 +55,471 @@ public void testMapConstructor() {
+	public void testTen() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		Set<Map.Entry<String, Serializable>> set = (Set<Map.Entry<String, Serializable>>) basicMapId.entrySet();
```
That test isn't really verifying the contract imposed by `Map` with testing just on an empty set. Not sure about the benefit, though.

TheRealHaui replied:
It tests that an empty entrySet gets returned.
I've renamed the test in addition.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java:
```diff
@@ -41,4 +55,471 @@ public void testMapConstructor() {
+	@Test(expected = IllegalArgumentException.class)
+	public void testSevenRaisesIllegalArgumentException() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId((Map<String, Serializable>) null);
```
Variable assignment never happens because of the raised exception.

TheRealHaui replied:
Variable declaration and assignment removed.
Only object instantiation left.

TheRealHaui commented:
Thank you for your response.
I have removed the commit regarding the README.md from this branch.
I'll respond to the others points via their specific reply fields.
Although, it won't be today anymore.

mp911de commented:
Any update on this PR?

TheRealHaui commented:
Sorry, didn't find time until now to integrate each and every points of your feedback.
I am planning this within the next two weeks.

TheRealHaui commented:
I have incorporated all the changes you requested.

TheRealHaui commented:
@mp911de 
I've corrected all issues remarked by you.
Furthermore I've added a bunch of new tests.
The builds do not fail because of my changes.

I suppose until your next feedback there's no formal action of mine needed here.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/mapping/SimpleUserTypeResolverTest.java:
```diff
@@ -0,0 +1,64 @@
+	@Test(expected = IllegalArgumentException.class)  //DATACASS-405
+	public void testObjectCreationKeySpaceNameMustNotBeNull() throws Exception {
+
+		Cluster cluster = Cluster.builder().addContactPoint("192.168.0.1").build();
```
Rather use mocks here to prevent accidental network interaction.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/mapping/SimpleUserTypeResolverTest.java:
```diff
@@ -0,0 +1,64 @@
+	@Test(expected = IllegalArgumentException.class)  //DATACASS-405
+	public void testObjectCreationKeySpaceNameMustContainText() throws Exception {
+
+		Cluster cluster = Cluster.builder().addContactPoint("192.168.0.1").build();
```
Rather use mocks here to prevent accidental network interaction.

mp911de commented on spring-data-cassandra/src/test/java/org/springframework/data/cassandra/mapping/SimpleUserTypeResolverTest.java:
```diff
@@ -0,0 +1,64 @@
+	@Test(expected = NoHostAvailableException.class)  //DATACASS-405
+	public void testResolveType() throws Exception {
+
+		Cluster cluster = Cluster.builder().addContactPoint("192.168.0.1").build();
```
Rather use mocks here to prevent accidental network interaction.

mp911de commented on spring-cql/src/test/java/org/springframework/cassandra/core/OrderingTest.java:
```diff
@@ -0,0 +1,98 @@
+
+		Ordering primaryKeyType = Ordering.valueOf("ASCENDING");
+
+		AssertionsForInterfaceTypes.assertThat(primaryKeyType).isEqualTo(Ordering.ASCENDING);
```
Use static imports for assertion methods.

mp911de commented on spring-cql/src/test/java/org/springframework/cassandra/core/OrderingTest.java:
```diff
@@ -0,0 +1,98 @@
+
+		Ordering primaryKeyType = Ordering.valueOf("DESCENDING");
+
+		AssertionsForInterfaceTypes.assertThat(primaryKeyType).isEqualTo(Ordering.DESCENDING);
```
Use static imports for assertion methods.

mp911de commented on spring-cql/src/test/java/org/springframework/cassandra/core/OrderingTest.java:
```diff
@@ -0,0 +1,98 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```
No blank line between license header and package declaration.

mp911de commented on spring-cql/src/test/java/org/springframework/cassandra/core/OrderingTest.java:
```diff
@@ -0,0 +1,98 @@
+
+
+	@Test
+	public void testCompareWithNonNullReturningPositive() throws Exception {
```
`test` prefix in methods isn't required since JUnit 4. Please express the goal of the test in the method name (such as `ascendingComparesLessThanToDescending`)

TheRealHaui commented:
@mp911de 
Thank you for your feedback.
I'll incoporate the changes.

However regarding point 2 I am not sure what's meant excactly.
Do you mean to verify within the test method if the exception was thrown or not or is it desired to delete the referring tests completely?

Regarding point 5: No problem, although never done.
I'll going to ask Google for it.

schauder commented:
Regarding point 2: 

Your test method signatures look like this:

	@Test
 	public void testCompareWithNullReturningPositive() throws Exception {

The `throws Exception` clause is only necessary if you do something inside the test which throws a checked exception. In all other tests the `throws Exception` clause should be removed, so it looks like this:

	@Test
 	public void testCompareWithNullReturningPositive() {

TheRealHaui commented:
@schauder 
Thanks for your answer.
Obviously, completely missunderstood what was meant.

mp911de commented:
Closing this pull request as it makes no longer sense to keep this long-lived pull request open.
